tfx_addons/example_filter/executor_test.py

```diff
@@ -0,0 +1,182 @@
+      module_file=evaluator_module_file,
+      examples=example_gen.outputs['examples'],
+      model=trainer.outputs['model'],
+      baseline_model=model_resolver.outputs['model'],
+      eval_config=eval_config)
+
+  pusher = tfx.extensions.google_cloud_ai_platform.Pusher(
+      model=trainer.outputs['model'],
+      model_blessing=evaluator.outputs['blessing'],
+      custom_config={
+          tfx.extensions.google_cloud_ai_platform.experimental.
+          PUSHER_SERVING_ARGS_KEY:
+          ai_platform_serving_args,
+      })
+
+  return tfx.dsl.Pipeline(
+      pipeline_name=pipeline_name,
+      pipeline_root=pipeline_root,
+      components=[
+          example_gen,
+          statistics_gen,
+          schema_gen,
+          example_validator,
+          trainer,
+          model_resolver,
+          evaluator,
+          pusher,
+      ],
+      enable_cache=True,
+      beam_pipeline_args=beam_pipeline_args,
+  )
+
+
+class ExampleFilterTest(test_case_utils.TfxTest):
+  def setUp(self):
+    super(ExampleFilterTest, self).setUp()
+    self.enter_context(test_case_utils.change_working_dir(self.tmp_dir))
+
+    self._penguin_root = os.path.dirname(__file__)
+
+    self._pipeline_name = 'sklearn_test'
+    self._data_root = os.path.join(self._penguin_root, 'data')
+    self._trainer_module_file = os.path.join(self._penguin_root,
+                                             'penguin_utils_sklearn.py')
+    self._evaluator_module_file = os.path.join(self._penguin_root,
+                                               'sklearn_predict_extractor.py')
+    self._pipeline_root = os.path.join(self.tmp_dir, 'tfx', 'pipelines',
+                                       self._pipeline_name)
+    self._ai_platform_training_args = {
+        'project': 'project_id',
+        'region': 'us-central1',
+    }
+    self._ai_platform_serving_args = {
+        'model_name': 'model_name',
+        'project_id': 'project_id',
+        'regions': ['us-central1'],
+    }
+
+  @mock.patch('tfx.components.util.udf_utils.UserModuleFilePipDependency.'
+              'resolve')
+  def testPipelineConstruction(self, resolve_mock):
+    # Avoid actually performing user module packaging because relative path is
+    # not valid with respect to temporary directory.
+    resolve_mock.side_effect = lambda pipeline_root: None
+
+    logical_pipeline = _create_pipeline(  # pylint:disable=protected-access
+        pipeline_name=self._pipeline_name,
+        pipeline_root=self._pipeline_root,
+        data_root=self._data_root,
+        trainer_module_file=self._trainer_module_file,
+        evaluator_module_file=self._evaluator_module_file,
+        ai_platform_training_args=self._ai_platform_training_args,
+        ai_platform_serving_args=self._ai_platform_serving_args,
+        beam_pipeline_args=[])
+
+    kubeflow_dag_runner.KubeflowDagRunner().run(logical_pipeline)
+    file_path = os.path.join(self.tmp_dir, 'sklearn_test.tar.gz')
```

Comment:
filename `'sklearn_test.tar.gz'` - is that correct?

Reply:
after some reevaluation I don't think that executor test is not needed and the tests can be done by component_test